### PR TITLE
Feature/edit user

### PR DESCRIPTION
**Pull request by KostetskyiYaroslav**
)))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))))
I'm calm at this moment
And you thought that I was doing nothing?)

### CODE REVIEW

File: MedicalDocManagment.WebUI/Controllers/api/AdminController.cs [hunk 1]
```diff
@@ -88,31 +88,56 @@ public IHttpActionResult GetPaged(int pageNumber = 1, int pageSize = 20)
...
             {
-                return NotFound();
-            }
-
-            //newPosition = _GetPositionById(userEditModel.Position.PositionId);
-
-            if (newPosition != null)
-            {
-                user.Position = newPosition;
+                return Request.CreateResponse(HttpStatusCode.NotFound, $"User wasn't found with ID {id}.");
```
Comment (AndriyKatsubo) [inline MedicalDocManagment.WebUI/Controllers/api/AdminController.cs]: Continuated line should be intended by four spaces.

File: MedicalDocManagment.WebUI/Controllers/api/AdminController.cs [hunk 2]
```diff
@@ -88,31 +88,56 @@ public IHttpActionResult GetPaged(int pageNumber = 1, int pageSize = 20)
...
+
+            return Request.CreateResponse(HttpStatusCode.OK);
+        }
+
+        private HttpResponseMessage _getErrorIdentityResult(IdentityResult result)
+        {
+
+            if (result == null)
+            {
+                return Request.CreateErrorResponse(HttpStatusCode.InternalServerError, "Error occured on the server");
```
Comment (AndriyKatsubo) [inline MedicalDocManagment.WebUI/Controllers/api/AdminController.cs]: Continuated line should be intended by four spaces.

File: MedicalDocManagment.WebUI/Controllers/api/AdminController.cs [hunk 3]
```diff
@@ -88,31 +88,56 @@ public IHttpActionResult GetPaged(int pageNumber = 1, int pageSize = 20)
...
+            if (result.Errors.Any())
+            {
+                StringBuilder errorsReport = new StringBuilder();
+
+                foreach (var error in result.Errors)
+                {
+                    errorsReport.Append(error + "; ");
+                }
+
+                return Request.CreateErrorResponse(HttpStatusCode.BadRequest, errorsReport.ToString());
```
Comment (AndriiKatsubo) [inline MedicalDocManagment.WebUI/Controllers/api/AdminController.cs:137]: Continuated line should be intended by four spaces.

File: MedicalDocManagment.WebUI/app/http.facade.ts [hunk 4]
```diff
@@ -12,33 +12,57 @@ import 'rxjs/add/operator/toPromise';
...
+    constructor(http: Http) {
+        this._http = http;
+    }
+
+    updateUser(user: UserModel) {
+        const jsonBody = JSON.stringify(user);
+        let headers = new Headers({ 'Content-Type': 'application/json;charset=utf-8' });
+
+        return this._http.put('/api/admin/edituser/' + user.id, jsonBody, { headers })
+            .map((resp: Response) => { return resp; })
```
Comment (AndriiKatsubo) [inline MedicalDocManagment.WebUI/app/http.facade.ts]: Please, correct this, so .map would begin under .put. Same for .catch.

File: MedicalDocManagment.WebUI/app/http.facade.ts [hunk 5]
```diff
@@ -12,33 +12,57 @@ import 'rxjs/add/operator/toPromise';
...
+            .catch((error: any) => { return Observable.throw(error); });
+    }
+
+    getPositionsList(): Observable<PositionModel[]> {
+        return this._http.get('/api/Admin/GetPositions')
+            .map((resp: Response) => resp.json())
```
Comment (AndriiKatsubo) [inline MedicalDocManagment.WebUI/app/http.facade.ts]: Please, correct this, so .map would begin under .get. Same for .catch.

File: MedicalDocManagment.WebUI/app/http.facade.ts [hunk 6]
```diff
@@ -12,33 +12,57 @@ import 'rxjs/add/operator/toPromise';
...
+            .catch((error: any) => { return Observable.throw(error); });
+    }
+
+    getUserById(id: string): Observable<UserModel> {
+        return this._http.get('/api/admin/getuser?id=' + id)
+            .map((resp: Response) => { return new UserModel(resp.json()); })
```
Comment (AndriiKatsubo) [inline MedicalDocManagment.WebUI/app/http.facade.ts]: Please, correct this, so .map would begin under .get. Same for .catch.

File: MedicalDocManagment.WebUI/app/http.facade.ts [hunk 7]
```diff
@@ -12,33 +12,57 @@ import 'rxjs/add/operator/toPromise';
...
+            .catch((error: any) => { return Observable.throw(error); })
+    }
+
+    getUsersList(): Observable<UsersModel> {
+        return this._http.get('/api/Admin/GetUsers')
+            .map((resp: Response) => { return new UsersModel(resp.json()); })
```
Comment (AndriiKatsubo) [inline MedicalDocManagment.WebUI/app/http.facade.ts]: Please, correct this, so .map would begin under .get. Same for .catch.

File: MedicalDocManagment.WebUI/app/http.facade.ts [hunk 8]
```diff
@@ -12,33 +12,57 @@ import 'rxjs/add/operator/toPromise';
...
+            .catch((error: any) => { return Observable.throw(error); });
+    }
+
+    getUsersListPaged(page: number, pageSize: number): Observable<PagedResponseModel> {
+        return this._http.get("api/admin/getpaged?pageNumber=" + page + "&pageSize=" + pageSize)
+            .map((resp: Response) => {
```
Comment (AndriiKatsubo) [inline MedicalDocManagment.WebUI/app/http.facade.ts]: Please, correct this, so .map would begin under .get. Same for .catch.

File: MedicalDocManagment.WebUI/app/users-list/page/item-actionlist-edit/modal/modal.component.ts [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+    positions: PositionModel[];
+
+    constructor(private _http: HttpFacade) { }
+
+    submit() {
+        this._http.updateUser(this.user)
+            .subscribe((data) => { console.log(data) });
+    }
+
+    ngOnInit() {
```
Comment (AndriiKatsubo) [inline MedicalDocManagment.WebUI/app/users-list/page/item-actionlist-edit/modal/modal.component.ts:24]: Consider placing ngOnInit() code block before submit() code block. 

> Placing members in a consistent sequence makes it easy to read and helps instantly identify which members of the component serve which purpose.

File: MedicalDocManagment.WebUI/app/users-list/page/item-actionlist-edit/button/button.component.ts [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+
+import UserModel from "../../../../models/usermodel";
+
+@Component({
+    moduleId: module.id,
+    selector: 'item-actionlist-edit-button',
+    templateUrl: 'button.component.html'
+})
+export default class ItemActionListEditButton {
+    @Input() user: UserModel = new UserModel();
```
Comment (AndriyKatsubo) [inline MedicalDocManagment.WebUI/app/users-list/page/item-actionlist-edit/button/button.component.ts]: Consider placing next alphabetized.

> Placing members in a consistent sequence makes it easy to read and helps instantly identify which members of the component serve which purpose.

File: MedicalDocManagment.WebUI/app/users-list/page/item-actionlist-edit/modal/modal.component.ts [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+import UserModel from "../../../../models/usermodel";
+import PositionModel from '../../../../models/positionmodel';
+
+@Component({
+    moduleId: module.id,
+    selector: 'item-actionlist-edit-modal',
+    templateUrl: 'modal.component.html'
+})
+export default class ItemActionListEditModal implements OnInit {
+    @Input() user: UserModel = new UserModel();
```
Comment (AndriyKatsubo) [inline MedicalDocManagment.WebUI/app/users-list/page/item-actionlist-edit/modal/modal.component.ts]: Consider placing next alphabetized.

> Placing members in a consistent sequence makes it easy to read and helps instantly identify which members of the component serve which purpose.

File: MedicalDocManagment.WebUI/app/users-list/page/item-actionlist.component.ts [hunk 12]
```diff
@@ -1,17 +1,27 @@
...
+    selector: '[item-ActionList]',
+    templateUrl: 'views/item-actionlist.component.html'
 })
 
 /**
  * Class, which implements list of actions on item from list of users.
  */
 export default class ItemActionListComponent {
-}
+    @Input() user: UserModel;
```
Comment (AndriiKatsubo) [inline MedicalDocManagment.WebUI/app/users-list/page/item-actionlist.component.ts]: Consider placing next alphabetized.

> Placing members in a consistent sequence makes it easy to read and helps instantly identify which members of the component serve which purpose.

File: MedicalDocManagment.WebUI/app/users-list/page/item.component.ts [hunk 13]
```diff
@@ -1,28 +1,33 @@
...
-  /**
-   * Property which receives data about user from users list.
-   * @type {UserModel}
-   */
-  @Input() user: UserModel;
+    /**
+     * Property which receives data about user from users list.
+     * @type {UserModel}
+     */
+    @Input() user: UserModel;
```
Comment (AndriyKatsubo) [inline MedicalDocManagment.WebUI/app/users-list/page/item.component.ts]: Consider placing next alphabetized.

> Placing members in a consistent sequence makes it easy to read and helps instantly identify which members of the component serve which purpose.

File: MedicalDocManagment.WebUI/app/users-list/page/page.component.ts [hunk 14]
```diff
@@ -1,28 +1,30 @@
...
+    templateUrl: 'views/page.component.html',
+    styleUrls: ['views/page.component.css'],
 })
-
 export default class PageComponent {
-  @Input() users: UsersModel;
-  @Input() page: number;
-  @Input() pageSize: number;
-  @Input() total: number;
+    @Input() users: UsersModel;
```
Comment (AndriyKatsubo) [inline MedicalDocManagment.WebUI/app/users-list/page/page.component.ts]: Consider placing next alphabetized.

> Placing members in a consistent sequence makes it easy to read and helps instantly identify which members of the component serve which purpose.

File: MedicalDocManagment.WebUI/app/users-list/users-list.component.ts [hunk 15]
```diff
@@ -9,44 +9,55 @@ import { HttpFacade } from '../http.facade';
...
-  getPage(page: number, pageSize: number) {
-      this.users = this._httpFacade.getUsersListPaged(page, pageSize)
-                       .do(data => {
-                           this.page = data.PageNumber;
-                           this.pageSize = data.PageSize;
-                           this.total = data.TotalRecordCount;
-                       })
-                       .map(data => { return data.Users;});
-  }
+    users: Observable<UsersModel>;
```
Comment (AndriiKatsubo) [inline MedicalDocManagment.WebUI/app/users-list/users-list.component.ts]: Consider placing next alphabetized.

> Placing members in a consistent sequence makes it easy to read and helps instantly identify which members of the component serve which purpose.

**Review by AndriiKatsubo — DISMISSED** [review]

File: MedicalDocManagment.WebUI/Controllers/api/AdminController.cs (hunk 3 above)
Comment (KostetskyiYaroslav) [reply MedicalDocManagment.WebUI/Controllers/api/AdminController.cs:137]: Yes, i know :)
What does your comment mean?

Comment (AndriiKatsubo) [reply MedicalDocManagment.WebUI/Controllers/api/AdminController.cs:137]: I am sorry, nevermind. 
When I was reviewing this line looked like it was broken into two lines with wrong format, but then I saw it again.

**Review by AndriiKatsubo — APPROVED** [review]

**Review by Ravennod — APPROVED** [review]

**Pull request merged**